### PR TITLE
DAOS-5935 dedup: Fix for dedup when csum disabled

### PR DESCRIPTION
**Pull request by ryon-jensen**
- Dedup wasn't being considered when csummer was
  initialized on the client when container open
  completed. Fixed by calling to cont_props_init.
- added dedup daos_tests to daos_core_test so
  dedup test failures will be caught by CI.
- The commandline option -U for dedup was missing
  for the getopt_long.

Signed-off-by: Ryon Jensen <ryon.jensen@intel.com>

### CODE REVIEW

**Review by vvenkates27 — DISMISSED** [review]

File: src/tests/ftest/daos_test/daos_core_test.yaml [hunk 1]
```diff
@@ -138,6 +138,10 @@ daos_tests:
...
       test_timeout: 220
     test_S:
-      daos_test: S 
+      daos_test: S
       test_name: DAOS rebuild ec tests
       test_timeout: 900
+    test_U:
+      daos_test: U
+      test_name: DAOS dedup tests
+      test_timeout: 900
```
Comment (jolivier23) [inline src/tests/ftest/daos_test/daos_core_test.yaml:147]: 900 seems a bit high here.  Rebuild is a special case.

**Review by jolivier23 — DISMISSED** [review]

File: src/tests/suite/daos_test.c [hunk 2]
```diff
@@ -368,7 +368,7 @@ main(int argc, char **argv)
 	memset(tests, 0, sizeof(tests));
 
 	while ((opt = getopt_long(argc, argv,
-				  "ampcCdtTVizxADKeoROg:n:s:u:E:f:Fw:W:hrNvbSl:",
+				"ampcCdtTVizUxADKeoROg:n:s:u:E:f:Fw:W:hrNvbSl:",
```
[the remark this answers is not in the document]
Comment (ryon-jensen) [reply src/tests/suite/daos_test.c:371]: Going to ignore this change request. It makes the line longer than 80

File: src/tests/ftest/daos_test/daos_core_test.yaml (hunk 1 above)
Comment (ryon-jensen) [reply src/tests/ftest/daos_test/daos_core_test.yaml:147]: good call

**Review by jolivier23 — DISMISSED** [review]

**Review by NiuYawei — APPROVED** [review]

**Comment by NiuYawei** [comment]
CI dedup tests failed because NVMe is enabled, you may need to skip the whole dedup test when NVMe is enabled, or improve the get_size() of dedup test and skip only verify (memcmp) mode dedup test cases.

**Review by jolivier23 — APPROVED** [review]

File: src/tests/suite/daos_dedup.c [hunk 3]
```diff
@@ -231,6 +245,13 @@ with_identical_updates(void *const *state, uint32_t iod_type, int csum_type,
 	daos_size_t		delta;
 	int			rc;
 
+	if (dedup_type == DAOS_PROP_CO_DEDUP_MEMCMP &&
+	    dedup_is_nvme_enabled(*state)) {
+		print_message("DAOS_PROP_CO_DEDUP_MEMCMP "
+			      "doesn't support NVMe.\n");
+		skip();
```
Comment (NiuYawei) [inline src/tests/suite/daos_dedup.c:252]: If you skip only the MEMCMP cases, the get_size() needs be improved to check NVMe free size when NVMe is enabled. (Seems the dedup threshold is larger than 4k in all dedup tests, so update will be landed to NVMe when NVMe is enabled, and we need to check NVMe free space respectively).

Comment (ryon-jensen) [reply src/tests/suite/daos_dedup.c:252]: okay, I'll just skip for all dedup for now. :-)

**Review by NiuYawei — DISMISSED** [review]

**Review by jolivier23 — APPROVED** [review]

**Comment by ryon-jensen** [comment]
mdtest_small failure: DAOS-5992
OSAOfflineParallelTest failure: DAOS-6019

**Review by jolivier23 — APPROVED** [review]

**Review by NiuYawei — DISMISSED** [review]

File: src/tests/suite/daos_test.c [hunk 4]
```diff
@@ -378,7 +378,7 @@ main(int argc, char **argv)
 	memset(tests, 0, sizeof(tests));
 
 	while ((opt = getopt_long(argc, argv,
-				  "ampcCdtTVizxADKeoROg:n:s:u:E:f:Fw:W:hrNvbSXl:",
+				  "ampcCdtTVizUxADKeoROg:n:s:u:E:f:Fw:W:hrNvbSXl:",
```
Comment (ashleypittman) [inline src/tests/suite/daos_test.c:381]: Is L336 correct, that seems to say --dedup is lowercase u (which is actually used twice).

Comment (ryon-jensen) [reply src/tests/suite/daos_test.c:381]: You're right. I'll fix it.

**Review by jolivier23 — DISMISSED** [review]

**Review by NiuYawei — APPROVED** [review]

**Comment by johannlombardi** [comment]
@ryon-jensen Could you please rebase the patch?

**Review by jolivier23 — APPROVED** [review]

**Comment by ryon-jensen** [comment]
Failure is DAOS-6033

**Review by NiuYawei — APPROVED** [review]

**Comment by ryon-jensen** [comment]
Rebuild EC failures: DAOS-6033
OSA Offline Parallel test: DAOS-6019

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**